### PR TITLE
[SnapshotSAS]Add Snapshot SAS

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by annatisch** [comment]
Thanks @xiafu-msft - the contents of the `_shared` submodule is intended to be identical between all three packages (blobs, files, queues).
We can certainly refactor the code in here, and move portions of it out of the shared module if you feel that would be better, but we should be able to copy and paste `_shared` between all 3 SDKs with no changes :)

**Comment by xiafu-msft** [comment]
Hi @annatisch 
Thanks for pointing this out!
Now shared_access_signature files under _shared are the same across those three packages. 
Please take a look at the path of the new files(eg. azure-sdk-for-python\sdk\storage\azure-storage-blob\azure\storage\**blob\shared_access_signature.py**) and make sure it's acceptable 😀

If the pull request luckily can be merged, would you like to squash it and merge 😀

File: sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 import six
 
-from ._shared.shared_access_signature import BlobSharedAccessSignature
+from azure.storage.blob.shared_access_signature import BlobSharedAccessSignature
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py]: Minor: Just a small stylistic tweak - could we move this to the bottom and use the form:
```python
from .shared_access_signature import BlobSharedAccessSignature
```

File: sdk/storage/azure-storage-blob/azure/storage/blob/shared_access_signature.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
+from azure.storage.blob._shared import sign_string, url_quote
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/shared_access_signature.py]: Could we please make this file private? (e.g. `_shared_access_signature.py`

File: sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py (hunk 1 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py]: Thanks for the suggestion! I've fixed it

File: sdk/storage/azure-storage-blob/azure/storage/blob/shared_access_signature.py (hunk 2 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/shared_access_signature.py]: absolutely! Renamed ^_^

**Pull request merged**